File: Data Structures & Algorithms/top-k-elements-in-list/submission-11.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let count = {};
        let freq = Array.from({length: nums.length + 1}, () => []);
        for(const n of nums)
        {
            count[n] = (count[n] || 0) + 1;
        }
        for (const n in count)
        {
            (freq[count[n]] = freq[count[n]] || []).push(n);
        }
        let res = []
        for(let i = freq.length - 1; i >= 0; i--)
        {
            for(const n of freq[i])
            {
                res.push(n);
                if(res.length == k)
                {
                    return res;
                }
            }
        }
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-11.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        i = 0
        result = []
        while i < len(nums):
            if i > 0:
                while i < len(nums) and nums[i - 1] == nums[i]:
                    i+=1
            l = i + 1
            r = len(nums) - 1
            while l < r:
                if nums[i] + nums[l] + nums[r] == 0:
                    result.append([nums[i], nums[l], nums[r]])
                    l += 1
                    while l < r and nums[l] == nums[l-1]:
                        l += 1
                elif nums[i] + nums[l] + nums[r] < 0:
                    l += 1
                    while l < r and nums[l] == nums[l-1]:
                        l += 1
                else:
                    r-=1
                    while l < r and nums[r] == nums[r + 1]:
                        r -= 1
            i+=1
        return result
                
